crates/steel-repl/src/repl.rs

```diff
@@ -233,13 +239,19 @@ pub fn repl_base(mut vm: Engine) -> std::io::Result<()> {
                     }
                     _ => {
                         // TODO also include this for loading files
-                        finish_or_interrupt(&mut engine.borrow_mut(), line, print_time);
+                        let now = Instant::now();
+
+                        finish_or_interrupt(&mut engine.borrow_mut(), line);
+
+                        if print_time {
+                            println!("Time taken: {:?}", now.elapsed());
+                        }
                     }
                 }
             }
             Err(ReadlineError::Interrupted) => {
                 println!("CTRL-C");
-                break;
+                continue;
```

Comment:
is the goal here to actually interrupt evaluation? I think if you do something like this:

```scheme
λ > (define (loop) (loop))
λ > (loop)
^C
➜  steel git:(repl)
```

You'll find that it exits the evaluation.

If you do `CTRL-C` while at the top level though, it won't exit though. Was that the goal? If you want it to interrupt evaluation and keep going, I think the changes will be slightly more involved - you'll have to register a handler for ctrl-c and catch it, and just don't exit and continue

Reply:
Oh, I foolishly assumed signals would _always_ be caught by readline. I'll look into adding a signal handler.

Reply:
Oh, wait, you'd need a way to _interrupt_ the VM execution. I see you had something like that a while ago (`vm.on_progress`). I think I could hack something together that works, but I worry that I won't be able to gauge whether a naive `break` in some instruction loop might leave the VM in an inconsistent state.

I'm happy to go either way: give it a try or leave `Ctrl-C` as is for now.

Reply:
I removed `on_progress` simply because I didn't dedicate enough time to see to how it would affect performance in the general case. If we did some benchmarking, I'd be happy to add it back in (perhaps behind a feature flag) - that way we also open the door for debuggers and other kinds of things

the easiest way to do this I think is to special case interruption, rather than use the general purpose `on_progress` - and just have the interpreter loop be a `while <not interrupted>` - and then in the interrupted case, return and error - the error should force all of the clean up that needs to be run.

There might be some edge cases with this, but that would be a nice direction to start if you're interested in contributing that